src/algebraic_geometry/projective_spectrum/scheme.lean

```diff
@@ -165,6 +166,114 @@ lemma mem_carrier.clear_denominator [decidable_eq (away f)]
       (∑ i in c.support.attach, acd (c i) (finset.mem_image.mpr ⟨i, ⟨i.2, rfl⟩⟩) * i.1.2.some) :=
 mem_carrier.clear_denominator' x $ (mem_carrier_iff 𝒜 x z).mpr hz
 
+
+section carrier'
+/--
+The underlying set of `to_Spec.carrier` is equal to the underlying set of ideal generated by
+elements in `A_f` whose numerator is in `x` and has the same degree as the denominator.
+-/
+def carrier' : ideal (A⁰_ f) :=
+ideal.span { z | ∃ ⦃s F : A⦄ (hs : s ∈ x.1.as_homogeneous_ideal) (n : ℕ)
+  (s_mem : s ∈ 𝒜 n) (F_mem1 : F ∈ 𝒜 n) (F_mem2 : F ∈ submonoid.powers f),
+  z = quotient.mk' ⟨_, ⟨s, s_mem⟩, ⟨F, F_mem1⟩, F_mem2⟩ }
+
+lemma carrier_eq_carrier' :
+  carrier 𝒜 x = carrier' 𝒜 x :=
+begin
```

Comment:
This proof is ~90 lines long. Are there some API sublemmas that can be extracted?